tests/unit/lib/charms/sdcore_nms_k8s/v0/test_fiveg_core_gnb_provider_interface.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+import json
+from unittest.mock import patch
+
+import pytest
+import scenario
+
+from tests.unit.lib.charms.sdcore_nms_k8s.v0.dummy_fiveg_core_gnb_provider_charm.src.dummy_provider_charm import (  # noqa: E501
+    DummyFivegCoreGnbProviderCharm,
+    PLMNConfig,
+)
+
+TEST_TAC_VALID = 1
+TEST_TAC_INVALID = 0
+TEST_MCC = "001"
+TEST_MNC = "01"
+TEST_SST = 1
+TEST_SD_VALID = 2
+TEST_SD_INVALID = "ssss"
+
+
+class TestFivegCoreGnbProviderCharm:
+    @pytest.fixture(autouse=True)
+    def context(self):
+        self.ctx = scenario.Context(
+            charm_type=DummyFivegCoreGnbProviderCharm,
+            meta={
+                "name": "fiveg-core-gnb-provider",
+                "provides": {
+                    "fiveg_core_gnb": {
+                        "interface": "fiveg_core_gnb",
+                    }
+                },
+            },
+            actions={
+                "publish-gnb-config": {
+                    "params": {
+                        "relation-id": {
+                            "description": "The relation ID of the relation to get the URL from",
+                            "type": "string",
+                        },
+                        "tac": {
+                            "type": "string",
+                        },
+                        "plmns": {
+                            "type": "string",
+                        },
+                    },
+                },
+            },
+        )
+
+    @pytest.fixture(autouse=True)
+    def setUp(self, request):
+        yield
+        request.addfinalizer(self.tearDown)
+
+    def tearDown(self) -> None:
+        patch.stopall()
+
+    @pytest.mark.parametrize("tac", [TEST_TAC_VALID, TEST_TAC_INVALID])
+    def test_given_fiveg_core_gnb_relation_when_publish_gnb_config_then_data_is_in_application_databag(  # noqa: E501
+        self, tac,
+    ):
+        fiveg_core_gnb_relation = scenario.Relation(
+            endpoint="fiveg_core_gnb",
+        )
+        state_in = scenario.State(
+            leader=True,
+            relations={fiveg_core_gnb_relation},
+        )
+
+        plmns = [PLMNConfig(mcc=TEST_MCC, mnc=TEST_MNC, sst=TEST_SST, sd=TEST_SD_VALID)]
+        params = {
+            "relation-id": str(fiveg_core_gnb_relation.id),
+            "tac": str(tac),
+            "plmns": json.dumps([plmn.asdict() for plmn in plmns])
+        }
+
+        if tac == TEST_TAC_INVALID:
```

Comment:
I would avoid having conditions inside the test and I would create 2 separate tests instead. Maybe it makes more sense to use the `parameterize` to group the invalid values test cases